### PR TITLE
feature (form): Email validation with error message

### PR DESCRIPTION
**Pull request by Adonsio**
@plentymarkets/team-terra

#### Must be done by Terra

Testing

-   [ ] Person 1 (mandatory)

    Browser support (relevant for changes in scss / appearance of components)

    -   [ ] Chrome
    -   [ ] Firefox
    -   [ ] Safari

    Responsiveness

    -   [ ] Desktop
    -   [ ] Tablet
    -   [ ] Mobile

-   [ ] Person 2 (mandatory)

    Browser support

    -   [ ] Chrome
    -   [ ] Firefox
    -   [ ] Safari

    Responsiveness

    -   [ ] Desktop
    -   [ ] Tablet
    -   [ ] Mobile

Information transfer

-   [ ] Inform Ceres about changes in `TerraFormComponent` (optional: relevant for changes in this component)

Terra Basic Plugin

-   [ ] Adapt changes (optional: relevant for version updates, global design changes, l10n)

---

#### Must be done by every developer

Please inform a member of Terra (@plentymarkets/team-terra) to get the upper part of the checklist done (with urgency or deadline).

Documentation

-   [ ] Changelog (optional: relevant for every change which influences the API)
-   [ ] Example (updated or created)
-   [ ] JSDoc (optional: relevant for every change which influences the API)
-   [ ] [Google spreadsheet](https://docs.google.com/spreadsheets/d/1OINnux8TEoitV-qdAxqUQaf9oGI_fqFVwfRWenRFfhI/edit#gid=0) (relevant when deprecating public APIs or features)

Testing

-   [ ] Unit Test (updated or created)

### CODE REVIEW

File: src/lib/components/forms/form/form-components/text-input/text-input.component.ts [hunk 1]
```diff
@@ -64,6 +64,10 @@ export class TextInputComponent implements ControlValueAccessor, TextInputInterf
     @Input()
     public name: string = '';
 
+    /** @description **/
```
Comment (Adonsio) [inline src/lib/components/forms/form/form-components/text-input/text-input.component.ts:67]:
```suggestion
    /** @description  whether entered text should be an email. **/
```

**Comment by nappynapster** [comment]
ein beispiel in `src/lib/components/forms/form/example/form-fields.ts` wäre noch ganz nett

**Review by nappynapster — APPROVED** [review]

**Comment by niwolf** [comment]
Changelog noch 👍

File: src/lib/components/forms/form/form-components/text-input/text-input.component.spec.ts [hunk 2]
```diff
@@ -114,6 +114,40 @@ describe('TextInputComponent', () => {
         expect(await formField.isControlValid()).toBeFalsy();
     });
 
+    it('should have email validate email when email is true', async () => {
```
Comment (pweyrich) [inline src/lib/components/forms/form/form-components/text-input/text-input.component.spec.ts:117]:
```suggestion
    it('should check whether the entered text is a valid email when input `email` is true', async () => {
```

**Review by nappynapster — APPROVED** [review]

**Review by pweyrich — APPROVED** [review]

**Pull request merged**